### DISCUSSION TITLE
Python的break效果

### DISCUSSION THREAD

**Discussion by enoch-lightning** [1101-GS4719-C/Python-Community-GS4719-C#338] [General]
今天與同學討論，產生一個疑問：就是break如果放在巢狀迴圈中，到底會終止哪些迴圈。經由實驗可發現，break只會終止其屬於的迴圈。若break敘述屬於內層迴圈，則效果就是終止內層迴圈；若break敘述屬於外層迴圈，則效果就是終止外層迴圈。
實驗影片如下
[迴圈break敘述](https://youtu.be/sfvaayInY-I)

**Comment by enoch-lightning** [comment]
若故意將外層迴圈與內層迴圈的變數都設為i，結果也一樣。代表內層迴圈在運作時，其i會取代外層迴圈的i。示意圖如下。
![image](https://user-images.githubusercontent.com/86505454/149099626-475cddea-7b2a-473e-823b-a74464093215.png)

**Comment by xCk27x** [comment]
讚啦 我隨口問的問題馬上就有解答了
會提出這個問題是因為之前有在C語言裡的試過
但不確定break在python是不是也一樣

**Comment by BryanNee** [comment]
對！而且break只針對迴圈有效而不是判斷 像是if
